appium-tests/common/common.spec.js

```diff
@@ -228,11 +228,18 @@ describe('Contacts Android', function () {
     function checkSession(done) {
         if (failedToStart) {
             fail('Failed to start a session');
-            done();
         }
+        done();
```

Comment:
Ah, what the hell, you are right! This is bizarre: calling `fail` in a `beforeEach` _does not prevent the spec from executing_. The spec gets marked as failed correctly, but it still runs through the entire spec. It seems like this is a known (and open) issue in Jasmine: https://github.com/jasmine/jasmine/issues/577

Grrrr, that's annoying. I wanted to clean the test code up by moving the session check into a `beforeEach` :(. Sigh...